client/components/data/query-post-counts/index.jsx

```diff
@@ -1,58 +1,35 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestPostCounts } from 'state/posts/counts/actions';
 import { isRequestingPostCounts } from 'state/posts/counts/selectors';
 
-class QueryPostCounts extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId && this.props.type === nextProps.type ) {
-			return;
-		}
-
-		this.request( nextProps );
-	}
+export default function QuerySiteDomains( { siteId, type } ) {
+	const requesting = useSelector( state => isRequestingPostCounts( state, siteId, type ) );
+	const dispatch = useDispatch();
+	const previousId = useRef( undefined );
+	const previousType = useRef( undefined );
 
-	request( props ) {
-		if ( props.requesting ) {
-			return;
+	useEffect( () => {
+		if ( ! requesting && siteId !== previousId.current && type !== previousType.current ) {
+			requestPostCounts( siteId, type )( dispatch );
```

Comment:
Does this look like the right way of using action builders that return a function with a `dispatch` parameter? (Is this something that `connect` handles automatically?)

Reply:
I think this is something redux-thunk handles for us. I'd expect this to just be `dispatch( requestPostCounts( siteId, type ) )` Does that work though?

Reply:
Yes, you should just `dispatch` that action like any other. The thunkiness is an implementation detail, doesn't change how the action should be used.

Reply:
Thank you! I didn't realise this was due to `redux-thunk`. Dispatching normally works nicely 👍